src/groth16/prover.rs

```diff
@@ -263,19 +251,19 @@ where
             let mut c =
                 EvaluationDomain::from_coeffs(std::mem::replace(&mut prover.c, Vec::new()))?;
 
-            a.ifft(&worker, &mut fft_kern)?;
-            a.coset_fft(&worker, &mut fft_kern)?;
-            b.ifft(&worker, &mut fft_kern)?;
-            b.coset_fft(&worker, &mut fft_kern)?;
-            c.ifft(&worker, &mut fft_kern)?;
-            c.coset_fft(&worker, &mut fft_kern)?;
+            a.ifft(&worker, fft_kern.get())?;
+            a.coset_fft(&worker, fft_kern.get())?;
+            b.ifft(&worker, fft_kern.get())?;
+            b.coset_fft(&worker, fft_kern.get())?;
+            c.ifft(&worker, fft_kern.get())?;
+            c.coset_fft(&worker, fft_kern.get())?;
```

Comment:
Instead of having to do this `.get()` everywhere you could implement `std::ops::DerefMut` on `fft_kern`